packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx

```diff
@@ -138,14 +139,22 @@ const FilterConfiguration: FC<Props> = ({
             setDraftFilterRule((oldFilterRule) => {
                 // TODO: Maybe this isn't the best place to do this.
                 // All this says is if a filter *was* disabled and had no
-                // value but now has a value, enable it. This is a way of
-                // keeping disabled and 'no value' in sync.
-                return oldFilterRule &&
-                    !oldFilterRule?.values?.length &&
-                    oldFilterRule?.disabled &&
-                    newFilterRule.values?.length
-                    ? { ...newFilterRule, disabled: false }
-                    : newFilterRule;
+                // value but now has a value, enable it. Also enable it if
+                // The operator requires no value (null/not null)
+                // This is a way of keeping disabled and 'no value' in sync.
+                let isNewFilterDisabled = newFilterRule.disabled;
```

Comment:
this logic doesn't have permissions into account, so this can also be done by viewers, right ?

Reply:
Good find. I think there is an existing bug where viewers can change this. They probably shouldn't be able to. This fix doesn't change the permissions though. 

I'm inclined to check this in as is and fix that separately if that's cool with you.